### ISSUE TITLE
git gui fails with "No working directory ..."

### ISSUE THREAD

**Issue by filnet** [Alexpux/MSYS2-packages#801]
Hi,

Since upgrading, `git gui` fails with this error popup:

![gitguierror](https://cloud.githubusercontent.com/assets/8101062/22463845/3e574096-e7b4-11e6-9826-f880a84781f4.png)

**Comment by filnet** [comment]
downgrading with

`pacman -U /var/cache/pacman/pkg/git-2.10.2-1-x86_64.pkg.tar.xz`

removes the issue.

**Comment by Fazer2** [comment]
Since the OP didn't specify the affected version, I'll add that this issue is still present in package git 2.12.1-1.

Before you get to this error, you will also stumble upon "bitmap "/usr/share/git-gui/lib/git-gui.ico" not defined".
It can be worked around with adding to your ~/.bashrc:
`export GIT_GUI_LIB_DIR=<msys2_installation_folder>/usr/share/git-gui/lib`
for instance:
`export GIT_GUI_LIB_DIR=/c/msys64/usr/share/git-gui/lib`
and then reloading your bash profile:
`. ~/.bashrc`

(based on https://sourceforge.net/p/msys2/discussion/general/thread/1bf2e8a6/)

**Comment by Fazer2** [comment]
This is still an issue with git 2.13.0-1.

**Comment by baldurk** [comment]
I ran into this today as well. I think it goes back to [this change](http://repo.or.cz/git-gui.git?a=commit;h=577c7e8fc663bc0e31b10e8691f03c3361dedc51) upstream, which prevents the code from running through cygpath to convert unix paths to windows paths.

Since I'm running through mingw64 for the above tcl/tk fixes, `$MSYSTEM` is `MINGW64` for me.

**Comment by Fazer2** [comment]
@baldurk Nice find. When I run `MSYSTEM=MSYS git gui`, the error from the original comment doesn't appear, but a new one does (while being in folder /d/dev/project/MINGW-packages/):
```
fatal: Invalid path '/d/dev/project/MINGW-packages/D:': No such file or directory
fatal: Invalid path '/d/dev/project/MINGW-packages/D:': No such file or directory
    while executing
"close $fd"
    (procedure "git" line 5)
    invoked from within
"git mktree << {}"
    (procedure "PARENT" line 9)
    invoked from within
"PARENT"
    (procedure "rescan_stage2" line 30)
    invoked from within
"rescan_stage2 file3faa610 ui_ready"
```
Git Gui window appears in the background and at first appears to be working, but doesn't find any modifications in working directory. When you try to open Repository -> Visualize All Branch History, it shows an error dialog with `Cannot find git repository here`.

**Comment by wrightsg** [comment]
This is still an issue with git 2.15.0.

Furthermore, the above workaround (setting GIT_GUI_LIB_DIR) does not work for me. Starting git-gui from gitk will result in a "Git directory not found: <path_to_repo>/.git".

Any update on this?

**Comment by Alexpux** [comment]
You need first cd to directory with git repo. But in any case you will get other errors

**Comment by wrightsg** [comment]
You are correct. I have to change to the root directory of the git repo (I was in a subdirectory previously). I now get "No working directory <path_to_repo>: couldn't change working directory to <path_to_repo>: no such file or directory". Is this what you expected? Can/will this be fixed?

**Comment by Alexpux** [comment]
Last error becasue git is MSYS2 program and works with UNIX paths, while  tcl/tk is MINGW program and it doesnt understandd UNIX paths

**Comment by ArekPiekarz** [comment]
@wrightsg What solved the problem for me was adding "Git for Windows" repository to Msys2 and installing Git from there. All the tools like Git Gui and Gitk work correctly this way. The guide how to do it is here - https://github.com/git-for-windows/git/wiki/Install-inside-MSYS2-proper

**Comment by Alexpux** [comment]
I'm widely use ````tig```` console program to work with git

**Comment by SlySven** [comment]
This still seems to be occurring in 2.26.2 two year after the last report - what needs to be done to fix the MSYS2 native package so that the pretty universal (well, ***I*** use them on **three** related GNU/Linux distributions and FreeBSD as *well*) tools `git gui` and `gitk` work without having to faff around with with foreign packages from https://github.com/git-for-windows ? :worried:

**Comment by kevinpowell** [comment]
This is still an issue in 2.28.0 -- I just ran into it today.  There is an acceptable way to get things working:

1. install the tk package for mingw (I used 64bit:  mingw-w64-x86_64-tk)
2. start mingw64 shell
3. GIT_GUI_LIB_DIR=/c/msys64/usr/share/git-gui/lib MSYSTEM=MSYS git gui

This still seems like something has gone wrong.  I don't want to globally set MSYSTEM=MSYS in a mingw64 shell.  Instead I guess I'll make an alias for git-gui that sets MSYSTEM.  It will be acceptable for now, as I don't use 'git gui' very much.  gitk (which I use more often) works without any weird env variables.

**Comment by pcopissa** [comment]
This is still an issue in 2.32

The workaround proposed above by @kevinpowell will allow `git gui` to start but that's about it. Other errors crop up down immediately when attempting to clone a repository, when trying to create one etc...
As of July 2021 `git gui` is unuseable out of the box.

**Comment by wswind** [comment]
Just meet the same problem

**Comment by Yoshqu** [comment]
yep, still need to add export GIT_GUI_LIB_DIR to make it work.

**Comment by pgpbpadilla** [comment]
Still true in 2.34.1-1 installed with `pacman` + `msys2`

Both workarounds work:
- setting the LIB env-var
- aliasing git to the binary installed from https://gitforwindows.org/

**Comment by HunterZ** [comment]
Happy 5th birthday to this issue! It's still broken with git 2.35.1.

`GIT_GUI_LIB_DIR=/usr/share/git-gui/lib git gui` isn't sufficient, because I still get working directory errors.

`GIT_GUI_LIB_DIR=/usr/share/git-gui/lib MSYSTEM=MSYS git gui` seems to work, but I have misgivings about it after reading other people's comments here. It also failed to clone a GitHub repo, but I never use it for that anyway.

**Comment by HunterZ** [comment]
This seems to have been fixed via pull request #2886, but others should probably verify.
